plugins/Vectorscope/VectorView.cpp

```diff
@@ -0,0 +1,284 @@
+/* VectorViewView.cpp - implementation of VectorViewView class.
+ *
+ * Copyright (c) 2019 Martin Pavelek <he29/dot/HS/at/gmail/dot/com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "VectorView.h"
+
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <QImage>
+#include <QPainter>
+
+#include "GuiApplication.h"
+#include "MainWindow.h"
+
+
+VectorView::VectorView(VecControls *controls, LocklessRingBuffer<sampleFrame> *inputBuffer, unsigned short displaySize, QWidget *parent) :
+	QWidget(parent),
+	m_controls(controls),
+	m_inputBuffer(inputBuffer),
+	m_bufferReader(*inputBuffer),
+	m_displaySize(displaySize),
+	m_persistTimestamp(0),
+	m_oldHQ(m_controls->m_highQualityModel.value()),
+	m_oldX(m_displaySize / 2),
+	m_oldY(m_displaySize / 2)
+{
+	setMinimumSize(200, 200);
+	setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+
+	connect(gui->mainWindow(), SIGNAL(periodicUpdate()), this, SLOT(periodicUpdate()));
+
+	m_displayBuffer.resize(sizeof qRgb(0,0,0) * m_displaySize * m_displaySize, 0);
+
+	#ifdef VEC_DEBUG
```

Comment:
Interesting, I'm not sure if I saw that before. I usually treat preprocessor stuff just as any other code and indent according to its function, so that the indentation a) does not break the flow of surrounding code and b) separates the conditional code so that, well, it is obvious that it runs only under certain conditions.

Reply:
There are several ways of indenting such directives.
- Never indent it
- Indent it except the `#` at the beginning
- Indent it, but don't indent anything in the surrounded block
- Treat it as plain code fragment